### PR TITLE
SAK-43377 - Centralize some kernel comparators for sorting consistency.

### PR DESCRIPTION
**Pull request by mpellicer**

### CODE REVIEW

File: kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/AliasCreatedTimeComparator.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+package org.sakaiproject.util.comparator;
+
+import java.util.Comparator;
+
+import org.sakaiproject.alias.api.Alias;
+
+public class AliasCreatedTimeComparator implements Comparator<Alias> {
+    public int compare(Alias lhs, Alias rhs) {
+    // Sort by date, then by ID to assure consistent order.
+    return lhs.getCreatedTime().compareTo(rhs.getCreatedTime()) * 10 + lhs.getId().compareTo(rhs.getId());
```
Comment (jesusmmp) [inline kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/AliasCreatedTimeComparator.java]: Spacing

**Comment by jesusmmp** [comment]
LGTM!

File: kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/AliasCreatedTimeComparator.java (hunk 1 above)
Comment (mpellicer) [reply kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/AliasCreatedTimeComparator.java]: Fixed. thanks

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/AuthzGroupIdComparator.java  1
- kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/RoleIdComparator.java  1
- kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/UserSortNameComparator.java  2
- kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/ResourceTypeLabelComparator.java  1
- kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/GroupTitleComparator.java  1
- kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/AliasIdComparator.java  1
- kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/AliasCreatedTimeComparator.java  1
- kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/ToolTitleComparator.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=5153381&bid=16925491)

**Comment by ern** [comment]
Nice idea Miguel!!!

Maybe a few unit tests might make some sense?

**Pull request merged**